contracts/FragToken.sol

```diff
@@ -64,20 +73,40 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable {
 
         // add to current locked amount
         _locksAmount[msg.sender] = _locksAmount[msg.sender] + amount;
-        _locksBlock[msg.sender] = block.number;
+        _locksBlock[msg.sender] = block.timestamp;
+
+        if(period == Period.TwoWeeks)
+            _lockTime[msg.sender] = block.timestamp + (2 * _TIMELOCK);
+        
+        if(period == Period.OneMonth)
+            _lockTime[msg.sender] = block.timestamp + (4 * _TIMELOCK);
+
+        if(period == Period.ThreeMonths)
+            _lockTime[msg.sender] = block.timestamp + (12 * _TIMELOCK);
+
+        if(period == Period.SixMonths)
+            _lockTime[msg.sender] = block.timestamp + (24 * _TIMELOCK);
+
+        if(period == Period.OneYear)
+            _lockTime[msg.sender] = block.timestamp + (52 * _TIMELOCK);
 
         transfer(address(this), amount);
 
         // We need to propagate the signature because it's the only reliable way to fetch the public key
         // of the sender from other chains.
         // emit total amount of locked tokens
-        emit Lock(msg.sender, signature, _locksAmount[msg.sender]);
+        emit Lock(msg.sender, signature, _locksAmount[msg.sender], _lockTime[msg.sender]);
     }
 
     function unlock(bytes calldata signature) external {
         require(
             block.number > _locksBlock[msg.sender] + _lockCooldown,
-            "Lock didn't expire"
+            "Lock cooldown didn't expire"
```

Comment:
actually cooldown should be removed no? we just need to use lock now

Reply:
oh well, yes of course!